packages/loopring_v3/contracts/amm/libamm/AmmStatus.sol

```diff
@@ -80,6 +77,8 @@ library AmmStatus
                 uint96(AmmData.POOL_TOKEN_MINTED_SUPPLY),
                 new bytes(0)
             );
+        } else {// reused pool
+            require(S.accountID == config.accountID, "INCONSISTENT_ACCOUNT_ID");
```

Comment:
@Brechtpd , what if `exchange.getDepositContract()` changed? In such case, can we simply do the following again or the pool can no longer be reused?
```solidity
S.allowance[address(this)][address(exchange.getDepositContract())] = type(uint256).max;
            exchange.deposit(
                address(this), // from
                address(this), // to
                address(this), // token
                uint96(AmmData.POOL_TOKEN_MINTED_SUPPLY),
                new bytes(0)
            );
```